Design/Workbench/design-arch.md

```diff
@@ -0,0 +1,241 @@
+
+### Plugins
+
+This subpackage will contain plugins such as the algorithm dialogs. This will be loaded on demand the first time the InterfaceFactory is requested to create a dialog. This should again reduce the loading overhead.
+
+### Instrument View/Slice Viewer/Spectrum Viewer...
+
+These will be separate libraries so that for example another application could embed the instrument view widget. The DLLs internal to those subpackages will only be loaded when that subpackage is imported to reduce
+the loading overhead and to not make users pay for features they do not use.
+
+### Reduction Gui
+
+This will contain the `reduction_gui` framework currently used by many customized interfaces.
+
+### Scripting
+
+This will contain the code related to executing scripts including a "runner" class that can execute arbitrary code, a code editor plus an in-process Qt IPython console.
+
+## Plotting
+
+`matplotlib` will provide the plotting abilities within the new workbench. It exposes all classes as an object-oriented API but also provides a procedural, state-machine api, `pyplot` interactive programming.
+
+In `pyplot` matplotlib provides a default window/toolbar that provides basic tools. This will be inadequate for our users so we will develop a custom toolbar for our the 1D, 2D and 3D plot types. We will provide a `cli` module
+that mimics the `pyplot` api but uses our defaults (configurable) for toolbars/window etc. Matplotlib uses a [tool][mpl-boilerplate] to generate their `pyplot` api. We can do something similar.
+
+By providing an alternate `cli` module we can preserve access to the standard `matplotlib.pyplot` module should anyone want access to this. A snippet of code such as:
+
+```python
+try:
+    import mantiqt.plotting.cli as plt
+except ImportError:
+    import matplotlib.pyplot as plt
+```
+
+would allow a script to function either using our cli or the built-in matplotlib one.
+
+### 1D Plots
+
+![1D Plot](1d-plot-window.png)
+
+### 2D Plots
+
+![2D Plot](2d-plot-window.png)
+
+These figures are meant to serve as a guide and will need to be tweaked as they are developed.
+
+### Figure Management
+
+It is desirable within the plotting framework to have a mechanism for controlling how figures are managed. For example, when `plot` is called without specifying a window then we must make a choice of either:
+
+ - create a new figure or
+ - merge/overplot with an existing figure.
+
+The initial propsosal is to borrow ideas from the [mslice][mslice] and have a controls on the figures so that users can indicate what is to be done with this window. In the figures above each figure contains a pair of
+buttons `Hold/Active` or `Held/Active` (depending on its state). These buttons control what happens when a plotting function is called without explicit intent as to which figure it should affect.
```

Comment:
@LamarMoore @NickDraper I feel this is something that, whatever we choose, it will be fine with some and not with others. I'm open to other suggestions if you have any.

I imagine initial user testing will give us some better feedback on this to react to.